pkg/awsutils/awsutils.go

```diff
@@ -1325,9 +1379,28 @@ func (cache *EC2InstanceMetadataCache) getFilteredListOfNetworkInterfaces() ([]*
 	networkInterfaces := make([]*ec2.NetworkInterface, 0)
 	for _, networkInterface := range result.NetworkInterfaces {
 		// Verify the description starts with "aws-K8S-"
-		if strings.HasPrefix(aws.StringValue(networkInterface.Description), eniDescriptionPrefix) {
-			networkInterfaces = append(networkInterfaces, networkInterface)
+		if !strings.HasPrefix(aws.StringValue(networkInterface.Description), eniDescriptionPrefix) {
+			continue
+		}
+		// Check that it's not a newly created ENI
+		tags := getTags(networkInterface, aws.StringValue(networkInterface.NetworkInterfaceId))
+
+		if value, ok := tags[eniCreatedAtTagKey]; ok {
+			parsedTime, err := time.Parse(time.RFC3339, value)
+			if err != nil {
+
+			}
```

Comment:
If we fail to parse the time here, should we just log a warning showing the tag, then tag it correctly?

Reply:
Overlooked this. Yeah I can do that.